corehq/apps/hqwebapp/static/hqwebapp/js/inactivity.js

```diff
@@ -167,6 +167,17 @@ hqDefine('hqwebapp/js/inactivity', [
                         $button.removeClass("btn-default").addClass("btn-danger");
                         $button.text(error);
                     } else {
+                        // Keeps the input value in the outer window in sync with newest token generated in
+                        // iframe_close_window after session timeout, avoiding csrf error.
+                        let iframe = $('iframe').get(0).contentWindow.document;
+                        let outerCSRFInput = $('#csrfTokenContainer');
+                        let iframeInputValue;
+                        try {
+                            iframeInputValue = iframe.getElementsByTagName('input')[0].value;
+                            outerCSRFInput.val(iframeInputValue);
+                        } catch(error) {
+                            return null;
```

Comment:
Is there a potential error you're anticipating?

What's the user experience if this happens? Are they able to close the popup?

Reply:
If for whatever reason there is an uncaught error finding and updating the elements the user won't be able to close the thank you popup. They'll have to refresh. I'm not sure how likely it is that this happens, but I was thinking it may be better to be overly cautious.

Reply:
Sounds reasonable. To make things a bit clearer for the user, can you show a message the way it's done a few lines up in the `if` block? It's not a great UX - the error message gets stuffed into the "Done" button - but it's at least feedback.